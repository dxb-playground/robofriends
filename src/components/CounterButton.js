import React, { PureComponent } from 'react';

class CounterButton extends PureComponent {
  constructor() {
    super();
    this.state = {
      count: 0,
    };
  }

  // shouldComponentUpdate(nextProps, nextState) {
  //   return this.state.count !== nextState.count;
  // }

  render() {
    return (
      <button color={this.props.color} onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }

  updateCount = () => {
    this.setState(state => ({ count: state.count + 1 }));
  }
}

export default CounterButton;
