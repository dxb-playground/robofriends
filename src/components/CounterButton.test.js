import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

it('expects to render the CounterButton component', () => {
  let color = 'red';
  let component = shallow(<CounterButton color={color} />);

  expect(component).toMatchSnapshot();
});

describe('clicking', () => {
  it('should increment count prop in state by one (1)', () => {
    let color = 'red';
    let component = shallow(<CounterButton color={color} />);

    // Below, we're grabbing the button in two (2) ways. One, with the id, and one
    // without.
    component.find('[id="counter"]').simulate('click');
    component.simulate('click');

    expect(component.state().count).toEqual(2);
    expect(component.props().color).toEqual('red');
  });
});
