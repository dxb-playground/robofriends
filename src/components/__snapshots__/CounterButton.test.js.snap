// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the CounterButton component 1`] = `ShallowWrapper {}`;
